inject/src/main/java/io/micronaut/inject/annotation/AnnotationMetadataWriter.java

```diff
@@ -413,6 +422,17 @@ public static void writeAnnotationDefaults(
                 }
 //                staticInit.visitLabel(falseCondition);
             }
+            if (annotationMetadata.repeated != null && !annotationMetadata.repeated.isEmpty()) {
+                Map<String, String> repeated = new HashMap<>();
```

Comment:
Should be linked to retain order and incremental compile?

Reply:
No, it's converted to a sorted string array